app/scripts/modules/core/src/delivery/details/singleExecutionDetails.component.ts

```diff
@@ -0,0 +1,142 @@
+                    analytics-event="Configuration"
+                    style="margin-right: 5px"
+                    uib-tooltip="Navigate to Pipeline Configuration"
+                    ui-sref="^.pipelineConfig({application: $ctrl.app.name, pipelineId: $ctrl.execution.pipelineConfigId})">
+              <span class="glyphicon glyphicon-cog"></span>
+              <span class="visible-md-inline visible-lg-inline">Configure</span>
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="row" ng-if="$ctrl.execution" >
+      <div class="col-md-10 col-md-offset-1 executions">
+        <execution execution="$ctrl.execution"
+                   application="$ctrl.app"
+                   standalone="true"
+                   show-stage-duration="$ctrl.executionFilterModel.showStageDuration"
+        ></execution>
+      </div>
+    </div>
+    <div class="row" ng-if="$ctrl.stateNotFound" style="min-height: 300px">
+      <h4 class="text-center">
+        <p>The execution cannot be found.</p>
+        <a ui-sref="^.executions.execution({application: $ctrl.app.name})">Back to Executions.</span>
+        </a>
+      </h4>
+    </div>
+  </div>
+`;
+
+class SingleExecutionDetailsController {
+  private executionScheduler: IScheduler;
+  private executionLoader: Subscription;
+
+  private execution: IExecution;
+  private stateNotFound: boolean;
+  private app: Application;
+
+  // Not sure if/where this is used.
+  public showDetails = () => true;
+
+  constructor(private schedulerFactory: SchedulerFactory,
+              private $state: StateService,
+              private executionService: ExecutionService,
+              public executionFilterModel: ExecutionFilterModel,
+              $scope: IScope) {
+    'ngInject';
+    $scope.$on('$stateChangeSuccess', (_event, _toState, toParams, _fromState, fromParams) => {
+      if (toParams.application !== fromParams.application || toParams.executionId !== fromParams.executionId) {
+        this.getExecution();
+      }
+    });
+  }
+
+  public $onInit() {
+    this.executionScheduler = this.schedulerFactory.createScheduler(5000);
+    this.executionLoader = this.executionScheduler.subscribe(() => this.getExecution());
+
+    this.getExecution();
+  }
+
+  public $onDestroy() {
+    this.executionScheduler.unsubscribe();
+    this.executionLoader.unsubscribe();
+  }
+
+  private getExecution() {
+    const { executionService, executionScheduler, executionLoader, $state } = this;
+
+    if (this.app.notFound) {
+      return;
+    }
+
+    executionService.getExecution($state.params.executionId).then((execution) => {
+      this.execution = this.execution || execution;
+      executionService.transformExecution(this.app, execution);
+      executionService.synchronizeExecution(this.execution, execution);
+      if (!execution.isActive) {
+        executionScheduler.unsubscribe();
+        executionLoader.unsubscribe();
```

Comment:
I just looked at the implementation and I have a suspicion that both calls are necessary to avoid leak.  @jrsquared ?